haystack/schema.py

```diff
@@ -101,11 +101,15 @@ def __init__(
         allowed_hash_key_attributes = ["content", "content_type", "score", "meta", "embedding"]
 
         if id_hash_keys is not None:
-            if not set(id_hash_keys) <= set(allowed_hash_key_attributes):
+            if not all(
+                True if key in allowed_hash_key_attributes or key.startswith("meta.") else False for key in id_hash_keys
+            ):
```

Comment:
This can be simplified like so, I think it's easier to read.

```suggestion
            if not all(key in allowed_hash_key_attributes or key.startswith("meta.") for key in id_hash_keys):
```